src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -1949,13 +1949,22 @@ export default class EthereumApi implements Api {
 
     const wallet = this.#wallet;
     const isKnownAccount = wallet.knownAccounts.has(fromString);
-    const privateKey = wallet.unlockedAccounts.get(fromString);
+    let privateKey = wallet.unlockedAccounts.get(fromString);
 
     if (privateKey === undefined) {
-      const msg = isKnownAccount
-        ? "authentication needed: passphrase or unlock"
-        : "sender account not recognized";
-      throw new Error(msg);
+      // if unlock all option is set to true
+      // then create a fake private key for unknown account
+      if(this.#options.wallet.unlockAll)
+      {
+        privateKey = wallet.createFakePrivateKey(fromString)
+      }
```

Comment:
Should this be implemented in the wallet, rather than the api? I can imagine us chasing different cases where we need to unlock the account.